ui/pages/settings/networks-tab/network-form/network-form.component.js

```diff
@@ -358,14 +358,27 @@ export default class NetworkForm extends PureComponent {
   };
 
   validateChainIdOnChange = (selfRpcUrl, chainIdArg = '') => {
+    const { t } = this.context;
     const { networksToRender } = this.props;
     const chainId = chainIdArg.trim();
+
     let errorKey = '';
     let errorMessage = '';
     let radix = 10;
-    const hexChainId = chainId.startsWith('0x')
-      ? chainId
-      : `0x${decimalToHex(chainId)}`;
+    let hexChainId = chainId;
+
+    if (!hexChainId.startsWith('0x')) {
+      try {
+        hexChainId = `0x${decimalToHex(hexChainId)}`;
+      } catch (err) {
+        this.setErrorTo('chainId', {
+          key: 'invalidHexNumber',
```

Comment:
The message `"Invalid Input"` may serve better for this case, but I wanted to avoid adding a new translation here, and I think serves the purpose well.